### ISSUE TITLE
Click column headers to 'sort by'

### ISSUE THREAD

**Issue by fsboehme** [Tudmotu/crowsnest#26]
(Just found this via your post in curious addys! Love it! Might be able to contribute a little.)

This should be easy and self-explanatory. I would like to be able to sort the table on any column by clicking its header. Click again to toggle between ascending/descending.

Cheers!

**Comment by Tudmotu** [comment]
Cool! 😃 Happy to see you here, Addys is definitely one of the cooler projects, and such a great mission!

This is definitely a needed feature, thank you for reporting it.

Feel free to try and take it, I can help with setting up the project and getting it to work if needed. And if you can't it's also fine! I will definitely implement it at some point.

**Comment by fsboehme** [comment]
got it up and running. what framework is that on the frontend? not recognizing it, but looks similar to react and vue, both of which i have some experience in.

**Comment by Tudmotu** [comment]
Not really any framework here. This started as a very basic HTML page and slowly evolved :smile: 

For this feature, I would start by looking at [`js/components/CollectionsTable.js`](https://github.com/Tudmotu/crowsnest/blob/main/js/components/CollectionsTable.js). That's what renders the table.
Nothing fancy there. Need to add event listener to the `.listHeader` elements and then sort the collections in the `render()` method.

**Comment by fsboehme** [comment]
in order to make this possible, `renderROIs` should probably live in `InvestmentState` or `CollectionsState` and store values on the objects, so we can simply sort on these values?

**Comment by fsboehme** [comment]
i guess `collection` – there's not actually an `investment` object

**Comment by Tudmotu** [comment]
Hmm, yes you are right. Thinking about it, not sure the InvestmentsState is even necessary, right?
We can add the investment stats directly inside the CollectionsState.
This might be a bigger change than I initially thought :sweat_smile: 

Is that what you meant?

**Comment by fsboehme** [comment]
yeah, exactly. probably makes sense to unify it in `CollectionsState`
